File: Estructura de Datos/Laboratorio2.py

# Crear dos matrices solicitando los datos por consola (cantidad de filas y columnas), y
# se debe añadir uno por uno los elementos de ambas matrices. Luego se deben sumar
# estas matrices en una función, y en otra función restarlas. Solo utilizando las funciones
# de la biblioteca numpy
print("EJERCICIO 1")
"importando biblioteca"
import numpy as np

#pedir al usuario que ingrese las columnas y filas de la 1ra matriz 
columnas1= int(input("ingrese numero de columnas de la Matriz 1: "))
filas1=int(input("ingrese numero de filas de la Matriz 1: "))

# Creamos la primera matriz y le pedimos al usuario que ingrese sus elementos
print("Ingrese los elementos de la primera matriz:")
matriz1 = np.array([list(map(int, input().split())) for _ in range(filas1)])

# Creamos la segunda matriz y le pedimos al usuario que ingrese sus elementos
print("Ingrese los elementos de la segunda matriz:")
matriz2 = np.array([list(map(int, input().split())) for _ in range(filas1)])

# Sumamos las dos matrices y almacenamos el resultado en una tercera matriz
matriz_suma = matriz1 + matriz2
print("la matriz suma es :", matriz_suma)

# Restamos las dos matrices y almacenamos el resultado en una tercera matriz
matriz_resta = matriz1 - matriz2
print("la matriz resta es :", matriz_resta)









# Crear dos matrices solicitando los datos por consola (cantidad de filas y columnas), y
# los elementos de estas matrices deben ser aleatorios del 1 al 5, no se deben ingresar
# por consola. Luego se deben sumar en una función las matrices, y en otra función
# restarlas. En este caso no se puede utilizar numpy, solo estructuras propias de Python.

import random
print("EJERCICIO 2")

#pedir al usuario que ingrese las columnas y filas de la 1ra matriz 
columnas1= int(input("ingrese numero de columnas de la Matriz 1: "))
filas1=int(input("ingrese numero de filas de la Matriz 1: "))


print("en la matriz 1 hay",columnas1, "columnas, y", filas1, "filas.") 

#Generar aleatoriamente los elementos de las dos matrices.
matriz1 = [[random.randint(0,5) for j in range(columnas1)] for i in range(filas1)]
matriz2 = [[random.randint(0,5) for j in range(columnas1)] for i in range(filas1)]
print("la matriz 1 se compone de: ",matriz1)
print("la matriz 2 se compone de: ",matriz2)

#ya genrados los elementos de ambas matrizes, las sumamos y guardamos el resultado en una matriz resultante
matriz_suma=[[matriz1[i][j] + matriz2[i][j] for j in range(columnas1)] for i in range(filas1)]
print("la matriz resultante de la suma es: ", matriz_suma)

#ya genrados los elementos de ambas matrizes, los restamos  y guardamos el resultado en una matriz resultante
matriz_resta=[[matriz1[i][j] - matriz2[i][j] for j in range(columnas1)] for i in range(filas1)]
print("la matriz resultante de la suma es: ", matriz_resta)

File: Estructura de Datos/laboratorio2.dart
// Crear tres listas:
// a. La primera lista debe tener los siguientes 5 elementos: [14,2,5,3,9]
// b. La segunda se debe ingresar por teclado (Debe tener 5 elementos
// enteros)
// c. La tercera lista igual debe contener 5 elementos, pero deben ser
// generados de forma aleatoria. (Solo números aleatorios negativos del
// -15 al -25)
// d. Restar la primera y segunda lista. La lista obtenida se suma con la
// tercera lista. Imprimir el resultado obtenido.
// e. Insertar en la primera y segunda posición de la lista que se obtuvo de la suma, los
// elementos 17 y 24.
// f. Por último ordenar de forma descendente la lista.


import 'dart:io';
import 'dart:math';

void main() {
  // a. Crear la primera lista con los siguientes elementos
  List<int> lista1 = [14, 2, 5, 3, 9];

  // b. La segunda se debe ingresar por teclado (Debe tener 5 elementos enteros)
  List<int> lista2 = [];
  print('Ingrese 5 elementos enteros para la segunda lista:');
  for (int i = 0; i < 5; i++) {
    int elemento = int.parse(stdin.readLineSync()!);
    lista2.add(elemento);
  }

  // c. La tercera lista igual debe contener 5 elementos, pero deben ser
// generados de forma aleatoria. (Solo números aleatorios negativos del
// -15 al -25)
  List<int> lista3 = [];
  Random random = Random();
  for (int i = 0; i < 5; i++) {
    int elemento = random.nextInt(11) - 15;
    lista3.add(elemento);
  }

  // d. Restar la primera y segunda lista, y sumarla con la tercera lista
  List<int> resultado = [];
  for (int i = 0; i < 5; i++) {
    int resta = lista1[i] - lista2[i];
    int suma = resta + lista3[i];
    resultado.add(suma);
  }

}



